### ISSUE TITLE
Big feature: generate C4 from docker-compose.yml and ansible

### ISSUE THREAD

**Issue by micheelengronne** [plantuml-stdlib/C4-PlantUML#120]
Hello,

I searched on the web some tools to extract a C4 diagram from docker-compose.yml and ansible collections.

Nothing convincing. Projects are not maintained and generate diagrams without standards.

Do you know any way to do that ?

Can this project do that ?

The idea would be to generate C4 diagrams automatically for doc purpose and to compare them to reference diagrams and discuss potential divergences between the plan and the implementation.

**Comment by Potherca** [comment]
Hi! It would be an **awesome** idea to create _with_ the C4 code-base, but not _in_ the C4 code-base, I think.

I can't speak for any of the other maintainers, but I would consider this feature to be rather far out of scope for this project.

A quick search turned up these libraries as most mentioned:

- https://github.com/funkwerk/compose_plantuml
- https://github.com/pmsipilot/docker-compose-viz

There also seem to be a lot of single-use scripts, like https://github.com/krogebry/awsmap/blob/master/lib/awsmap/mappers/docker_compose.py lying around. (And many more in Gists).

So it seems to be possible to build something yourself, the limit obviously being the amount of time and knowledge you have to spend, which is also the main reason I think it would be a poor fit for this project. docker-compose and  PlantUML (and C4) have more-or-less nothing in common :disappointed: 

However, if you are inclined to spend time on this, I am more than willing to offer advise or support.

**Comment by micheelengronne** [comment]
Yes, time is something I don't currently have, sadly. No worries, I will put it back in my backlog and try to make some progress on my epics for that.

**Comment by Potherca** [comment]
>  time is something I don't currently have, sadly

Very recognizable :grinning: 

Let's see what happens!

**Comment by kirchsth** [comment]
Hi  @micheelengronne,
I added properties to the standard (I plane something similar), maybe they are useful for you.
[Can you check the current implementation?](https://github.com/plantuml-stdlib/C4-PlantUML/blob/master/README.md#element-properties)
 Or if you need some extensions please inform me, maybe I have a simple solution that I can add it too?
BR
Helmut

PS.: I wrote a [structurz based generator](https://github.com/structurizr/dotnet-extensions/blob/master/docs/c4-plantuml.md) (which uses an older/my C4-Plantuml version). I want to update/extend it that it supports the new features too (properties, tags, ...) but you can easily extend the current version with your requirements too in the meantime too. 
The only remaining part would be that you have to write the required docker compose and ansible parser and create a C4 model.

**Comment by micheelengronne** [comment]
That's cool and from my POV a good step towards automation. For your DOTNET implementation I can not be of any help, I am on Linux and have never touched a Microsoft language.
